docs/README.md

```diff
@@ -900,15 +900,132 @@ where the schema for `CreateProductInput` is
 
 ### 2. Events
 
+Events are **immutable records of facts** within your application's domain. They are the cornerstone of Booster because of its event-driven and event-sourced nature. Booster events are TypeScript classes decorated with `@Event`. An event class may look like this:
+
+```typescript
+@Event
+export class EventName {
+  public constructor(readonly field1: SomeType,
+                     readonly field2: SomeOtherType,
+                     /* This event's entity ID must be present */) {}
+
+  public entityID(): UUID {
+    return /* the associated entity ID */
+  }
+}
+```
+
+Events and [entities](#4-entities-and-reducers) are closely related because entities represent the application's state after reducing the stream of events. All the events associated with the same entity are reduced together in the order they have occurred. If an event is related to an entity or not, it is domain-dependent. That's why Booster requires you to implement the `entityID` method. Entities and events are somehow coupled, but that coupling is limited to data. Take a look at this event:
+
+```typescript
+@Event
+export class CartPaid {
+  public constructor(
+    readonly cartID: UUID,
+    readonly paymentID: UUID) {}
+
+  public entityID(): UUID {
+    // returning cartID or paymentID depends on the business logic
```

Comment:
I thought the same but in [the example](https://github.com/boostercloud/booster/blob/master/packages/framework-integration-tests/src/events/CartPaid.ts#L9) of the integration test package, we are returning the paymentID. Actually, the confusion around this makes me write [an RFC](https://agilemonkeys.atlassian.net/wiki/spaces/B/pages/1267269684/Wrap+UUID+with+a+value+object+called+EntitynameID+when+generating+entities+to+avoid+ambiguity) about improving how we handle UUIDs to avoid ambiguity